From: =?utf-8?b?T3R0byBLZWvDpGzDpGluZW4=?= <otto@debian.org>
Date: Thu, 28 Nov 2024 21:29:03 -0800
Subject: Put build artifacts in `debian/.build/upstream` to avoid polluting
 upstream sources

Instead of putting build artifacts in `_build/` at the project root,
where they pollute the upstream sources and lure some people to also
modify the upstream `.gitignore` file, put it in the `debian/`
subdirectory along with all other Debian build artifacts.
---
 template.go | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/template.go b/template.go
index c589db4..593c4d5 100644
--- a/template.go
+++ b/template.go
@@ -86,6 +86,7 @@ func writeDebianGitIgnore(dir, debLib, debProg string, pkgType packageType) erro
 	fmt.Fprintf(f, "*.log\n")
 	fmt.Fprintf(f, "*.substvars\n")
 	fmt.Fprintf(f, "/.debhelper/\n")
+	fmt.Fprintf(f, "/.build/\n")
 	fmt.Fprintf(f, "/debhelper-build-stamp\n")
 	fmt.Fprintf(f, "/files\n")
 
@@ -298,7 +299,11 @@ func writeDebianRules(dir string, pkgType packageType) error {
 	fmt.Fprintf(f, "#!/usr/bin/make -f\n")
 	fmt.Fprintf(f, "\n")
 	fmt.Fprintf(f, "%%:\n")
-	fmt.Fprintf(f, "\tdh $@ --builddirectory=_build --buildsystem=golang\n")
+	fmt.Fprintf(f, "\tdh $@ --builddirectory=debian/.build/upstream --buildsystem=golang\n")
+	// Note: The above `--builddirectory=debian/.build/upstream` will eventually be obsolete
+	// in 2028+ then the dh-golang version 1.64+ that has merged
+	// https://salsa.debian.org/go-team/packages/dh-golang/-/merge_requests/26
+
 	if pkgType == typeProgram {
 		fmt.Fprintf(f, "\n")
 		fmt.Fprintf(f, "override_dh_auto_install:\n")
